### PR TITLE
[Php] Fix filter cache on PolyfillPackagesProvider

### PR DESCRIPTION
**Pull request by samsonasik**
Ref https://github.com/rectorphp/rector-src/pull/5388#pullrequestreview-1795601451

### CODE REVIEW

File: src/Php/PolyfillPackagesProvider.php [hunk 1]
```diff
@@ -50,7 +52,7 @@ public function provide(): array
      */
     private function filterPolyfillPackages(array $require): array
     {
-        return array_filter($require, static fn (string $packageName): bool => ! str_starts_with(
+        return array_filter(array_keys($require), static fn (string $packageName): bool => str_starts_with(
```
Comment (samsonasik) [inline src/Php/PolyfillPackagesProvider.php:55]: here only `key` is needed, when we define in `composer.json`

```php
"require": {
    "symfony/polyfill-php80" => "*"
}
```

then, filter the `str_starts_with()` one instead, so it get data:

```php
array('symfony/polyfill-php80')
```

to compare with `in_array()` on `PhpVersionedFilter`

https://github.com/rectorphp/rector-src/blob/6ebab94b971a930457c8e217dcb699c671ad617b/packages/VersionBonding/PhpVersionedFilter.php#L32-L37

File: src/Php/PolyfillPackagesProvider.php [hunk 2]
```diff
@@ -27,12 +27,14 @@ public function provide(): array
             return SimpleParameterProvider::provideArrayParameter(Option::POLYFILL_PACKAGES);
         }
 
-        $projectComposerJson = getcwd() . '/composer.json';
-        if (! file_exists($projectComposerJson)) {
-            return [];
+        // already cached, even only empty array
+        if ($this->cachedPolyfillPackages !== null) {
```
Comment (samsonasik) [inline src/Php/PolyfillPackagesProvider.php:31]: here compare no null = already cached.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

File: src/Php/PolyfillPackagesProvider.php (hunk 1 above)
Comment (samsonasik) [reply src/Php/PolyfillPackagesProvider.php:55]: I created e2e test for it:

- https://github.com/rectorphp/rector-src/pull/5392

**Comment by TomasVotruba** [comment]
Awesome, thanks for catching the glitches 👌

**Pull request merged**